### PR TITLE
[BPROC-418]-Remove buyer address when address is equal to pagarme or company

### PR DESCRIPTION
**Pull request by davidsmedeiros**
## Description

Este PR remove o campo endereço do buyer no request para a boleto-api quando o endereço é substituído pelo endereço do pagarme ou for o mesmo da company.

## Your checklist for this pull request

:rotating_light: Please review this items for a good pull request. :four_leaf_clover:

1. I've read the project's [Contributing Guidelines](../CONTRIBUTING.md)
1. My commits are well written and follow [pagarme/git-style-guide](https://github.com/pagarme/git-style-guide)
1. My changes are well covered by **tests** and **logs**
1. I've updated the project docs (if needed)
1. I fell safe about this implementation
1. I feel comfortable with the code I wrote, and I'm not ashamed to show it to my friends

In a good pull request, everything above is true :relaxed:

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5c2f64bd7b38cf2084400fe4ee6f7ffdda3239e8
  * Commits must be in the following format: **%{scope}: %{description}**

Check out our Git Style Guide here: https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 39429127e8d2d446da3ee36f548286affced4778
  * Commits must be in the following format: **%{scope}: %{description}**

Check out our Git Style Guide here: https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Review by leojasmim — APPROVED** [review]

File: test/unit/providers/boleto-api-caixa/index.js [hunk 1]
```diff
@@ -486,6 +490,73 @@ test('buildPayload complete', async (t) => {
   })
 })
 
+test('payer_address is different than company_address and pagarme address', async (t) => {
```
Comment (cersouza) [inline test/unit/providers/boleto-api-caixa/index.js:493]: Uma sugestão @davidsmedeiros, colocar no nome do teste o que esperado de retorno. Ex: "When payer_address is different than... **then should**...".

File: test/unit/providers/boleto-api-caixa/index.js [hunk 2]
```diff
@@ -486,6 +490,73 @@ test('buildPayload complete', async (t) => {
...
+    number: '308',
+    street: 'Rua dos bancos',
+    complement: '11º andar',
+    district: 'Brooklin',
+    city: 'São Paulo',
+    stateCode: 'SP',
+  })
+})
+
+test('payer_address is equals than company_address', async (t) => {
```
Comment (cersouza) [inline test/unit/providers/boleto-api-caixa/index.js:520]: Uma sugestão @davidsmedeiros, colocar no nome do teste o que esperado de retorno. Ex: "When payer_address is different than... **then should**...".

File: test/unit/providers/boleto-api-caixa/index.js [hunk 3]
```diff
@@ -486,6 +490,73 @@ test('buildPayload complete', async (t) => {
...
+    city: 'São Paulo',
+    state: 'SP',
+  }
+
+  const payload = buildPayload(boleto, operationId)
+
+  t.deepEqual(payload.buyer.address, undefined)
+})
+
+test('payer_address is equals than address pagarme', async (t) => {
```
Comment (cersouza) [inline test/unit/providers/boleto-api-caixa/index.js:549]: Uma sugestão @davidsmedeiros, colocar no nome do teste o que esperado de retorno. Ex: "When payer_address is different than... **then should**...".

**Review by cersouza — CHANGES_REQUESTED** [review]

File: src/providers/boleto-api-caixa/index.js [hunk 4]
```diff
@@ -114,6 +116,38 @@ const defineInterest = (boleto) => {
   return null
 }
 
+const buyerAdressIsEqualsToPagarme = (boleto) => {
```
Comment (brunoglm) [inline src/providers/boleto-api-caixa/index.js:119]: opa, @davidsmedeiros aqui da para retornarmos direto o resultado 😀 


```suggestion
const buyerAdressIsEqualsToPagarme = (boleto) => {
```
```suggestion
const buyerAdressIsEqualsToPagarme = (boleto) => (equals(boleto.payer_address, boleto.company_address))
```

Comment (brunoglm) [inline src/providers/boleto-api-caixa/index.js:119]: opa, @davidsmedeiros aqui da para retornarmos direto o resultado 😀 


```suggestion
const buyerAdressIsEqualsToPagarme = (boleto) => {
```
```suggestion
const buyerAdressIsEqualsToPagarme = (boleto) => (equals(boleto.payer_address, boleto.company_address))
```

File: src/providers/boleto-api-caixa/index.js [hunk 5]
```diff
@@ -114,6 +116,38 @@ const defineInterest = (boleto) => {
...
+  if (equals(boleto.payer_address, boleto.company_address)) {
+    return true
+  }
+  return false
+}
+
+const buyerAdressIsEqualsToAdressCompany = (boleto) => {
```
Comment (brunoglm) [inline src/providers/boleto-api-caixa/index.js:126]: opa, aqui também da para retornarmos direto igual a sugestão acima 😀

File: test/unit/providers/boleto-api-caixa/index.js [hunk 6]
```diff
@@ -406,6 +401,17 @@ test('buildPayload with interest per percentage and amount should return the int
 test('buildPayload complete', async (t) => {
   const operationId = cuid()
   const boleto = await createBoleto({ company_name: 'company name test' })
+
+  boleto.payer_address = {
```
Comment (brunoglm) [inline test/unit/providers/boleto-api-caixa/index.js:405]: opa, aqui da para passar o objeto direto no método `createBoleto` , igual feito nos testes abaixo 😀

Comment (brunoglm) [inline test/unit/providers/boleto-api-caixa/index.js:405]: opa, aqui da para passar o objeto direto no método `createBoleto` , igual feito nos testes abaixo 😀

**Review by cersouza — APPROVED** [review]
Boaa @davidsmedeiros 💥

**Review by brunoglm — APPROVED** [review]

**Pull request merged**